true
null
true
[32mPASS[m

false
null
false
[32mPASS[m

null
42
null
[32mPASS[m

1
null
1
[32mPASS[m

-1
null
-1
[32mPASS[m

{}
null
{}
[32mPASS[m

[]
null
[]
[32mPASS[m

{x: -1}
null
{"x": -1}
[32mPASS[m

.
Ôªø"byte order mark"
"byte order mark"
test input: Failed reading: not a valid json value context:[]
[31mERROR[m

"Aa\r\n\t\b\f\u03bc"
null
"Aa\u000d\u000a\u0009\u0008\u000c\u03bc"
[32mPASS[m

.
"Aa\r\n\t\b\f\u03bc"
"Aa\u000d\u000a\u0009\u0008\u000c\u03bc"
[32mPASS[m

"inter\("pol" + "ation")"
null
"interpolation"
[32mPASS[m

@text,@json,([1,.] | (@csv, @tsv)),@html,@uri,@sh,@base64,(@base64 | @base64d)
"<>&'\"\t"
"<>&'\"\t"
"\"<>&'\\\"\\t\""
"1,\"<>&'\"\"\t\""
"1\t<>&'\"\\t"
"&lt;&gt;&amp;&apos;&quot;\t"
"%3C%3E%26'%22%09"
"'<>&'\\''\"\t'"
"PD4mJyIJ"
"<>&'\"\t"
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

@base64
"fo√≥bar\n"
"Zm/Ds2Jhcgo="
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

@base64d
"Zm/Ds2Jhcgo="
"fo√≥bar\n"
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

@uri
"\u03bc"
"%CE%BC"
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

@html "<b>\(.)</b>"
"<script>hax</script>"
"<b>&lt;script&gt;hax&lt;/script&gt;</b>"
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[.[]|tojson|fromjson]
["foo", 1, ["a", 1, "b", 2, {"foo":"bar"}]]
["foo",1,["a",1,"b",2,{"foo":"bar"}]]
[32mPASS[m

{a: 1}
null
{"a":1}
[32mPASS[m

{a,b,(.d):.a,e:.b}
{"a":1, "b":2, "c":3, "d":"c"}
{"a":1, "b":2, "c":1, "e":2}
[32mPASS[m

{"a",b,"a$\(1+1)"}
{"a":1, "b":2, "c":3, "a$2":4}
{"a":1, "b":2, "a$2":4}
[32mPASS[m

%%FAIL
{(0):1}
jq: error: Cannot use number (0) as object key at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

%%FAIL
{non_const:., (0):1}
jq: error: Cannot use number (0) as object key at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

.foo
{"foo": 42, "bar": 43}
42
[32mPASS[m

.foo | .bar
{"foo": {"bar": 42}, "bar": "badvalue"}
42
[32mPASS[m

.foo.bar
{"foo": {"bar": 42}, "bar": "badvalue"}
42
[32mPASS[m

.foo_bar
{"foo_bar": 2}
2
[32mPASS[m

.["foo"].bar
{"foo": {"bar": 42}, "bar": "badvalue"}
42
[32mPASS[m

."foo"."bar"
{"foo": {"bar": 20}}
20
[32mPASS[m

[.[]|.foo?]
[1,[2],{"foo":3,"bar":4},{},{"foo":5}]
[3,null,5]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[.[]|.foo?.bar?]
[1,[2],[],{"foo":3},{"foo":{"bar":4}},{}]
[4,null]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[..]
[1,[[2]],{ "a":[1]}]
[[1,[[2]],{"a":[1]}],1,[[2]],[2],2,{"a":[1]},[1],1]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[.[]|.[]?]
[1,null,[],[1,[2,[[3]]]],[{}],[{"a":[1,[2]]}]]
[1,[2,[[3]]],{},{"a":[1,[2]]}]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[.[]|.[1:3]?]
[1,null,true,false,"abcdef",{},{"a":1,"b":2},[],[1,2,3,4,5],[1,2]]
[null,"bc",[],[2,3],[2]]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

try (.foo[-1] = 0) catch .
null
"Out of bounds negative array index"
test program: endOfInput context:[]
[31mERROR[m

try (.foo[-2] = 0) catch .
null
"Out of bounds negative array index"
test program: endOfInput context:[]
[31mERROR[m

.[-1] = 5
[0,1,2]
[0,1,5]
execution failed with this error:
_assign is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

.[-2] = 5
[0,1,2]
[0,5,2]
execution failed with this error:
_assign is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

.[]
[1,2,3]
1
2
3
[32mPASS[m

1,1
[]
1
1
[32mPASS[m

1,.
[]
1
[]
[32mPASS[m

[.]
[2]
[[2]]
[32mPASS[m

[[2]]
[3]
[[2]]
[32mPASS[m

[{}]
[2]
[{}]
[32mPASS[m

[.[]]
["a"]
["a"]
[32mPASS[m

[(.,1),((.,.[]),(2,3))]
["a","b"]
[["a","b"],1,["a","b"],"a","b",2,3]
[32mPASS[m

[([5,5][]),.,.[]]
[1,2,3]
[5,5,[1,2,3],1,2,3]
[32mPASS[m

{x: (1,2)},{x:3} | .x
null
1
2
3
[32mPASS[m

.[-2]
[1,2,3]
2
[32mPASS[m

[range(0;10)]
null
[0,1,2,3,4,5,6,7,8,9]
[32mPASS[m

[range(0,1;3,4)]
null
[0,1,2, 0,1,2,3, 1,2, 1,2,3]
[32mPASS[m

[range(0;10;3)]
null
[0,3,6,9]
[32mPASS[m

[range(0;10;-1)]
null
[]
[32mPASS[m

[range(0;-5;-1)]
null
[0,-1,-2,-3,-4]
[32mPASS[m

[range(0,1;4,5;1,2)]
null
[0,1,2,3,0,2, 0,1,2,3,4,0,2,4, 1,2,3,1,3, 1,2,3,4,1,3]
[32mPASS[m

[while(.<100; .*2)]
1
[1,2,4,8,16,32,64]
execution failed with this error:
while is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[(label $here | .[] | if .>1 then break $here else . end), "hi!"]
[0,1,2]
[0,1,"hi!"]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[(label $here | .[] | if .>1 then break $here else . end), "hi!"]
[0,2,1]
[0,"hi!"]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

%%FAIL
. as $foo | break $foo
jq: error: $*label-foo is not defined at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[.[]|[.,1]|until(.[0] < 1; [.[0] - 1, .[1] * .[0]])|.[1]]
[1,2,3,4,5]
[1,2,6,24,120]
execution failed with this error:
until is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[label $out | foreach .[] as $item ([3, null]; if .[0] < 1 then break $out else [.[0] -1, $item] end; .[1])]
[11,22,33,44,55,66,77,88,99]
[11,22,33]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[foreach range(5) as $item (0; $item)]
null
[0,1,2,3,4]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[foreach .[] as [$i, $j] (0; . + $i - $j)]
[[2,1], [5,3], [6,4]]
[1,3,5]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[foreach .[] as {a:$a} (0; . + $a; -.)]
[{"a":1}, {"b":2}, {"a":3, "b":4}]
[-1, -1, -4]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[limit(3; .[])]
[11,22,33,44,55,66,77,88,99]
[11,22,33]
[32mPASS[m

[limit(0; error)]
"badness"
[]
[32mPASS[m

[limit(1; 1, error)]
"badness"
[1]
[32mPASS[m

[first(range(.)), last(range(.)), nth(0; range(.)), nth(5; range(.)), try nth(-1; range(.)) catch .]
10
[0,9,0,5,"nth doesn't support negative indices"]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

first(1,error("foo"))
null
1
execution failed with this error:
first is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[limit(5,7; range(9))]
null
[0,1,2,3,4,0,1,2,3,4,5,6]
[32mPASS[m

[nth(5,7; range(9;0;-1))]
null
[4,2]
[32mPASS[m

[range(0,1,2;4,3,2;2,3)]
null
[0,2,0,3,0,2,0,0,0,1,3,1,1,1,1,1,2,2,2,2]
[32mPASS[m

[range(3,5)]
null
[0,1,2,0,1,2,3,4]
[32mPASS[m

[(index(",","|"), rindex(",","|")), indices(",","|")]
"a,b|c,d,e||f,g,h,|,|,i,j"
[1,3,22,19,[1,5,7,12,14,16,18,20,22],[3,9,10,17,19]]
[32mPASS[m

join(",","/")
["a","b","c","d"]
"a,b,c,d"
"a/b/c/d"
[32mPASS[m

[.[]|join("a")]
[[],[""],["",""],["","",""]]
["","","a","aa"]
[32mPASS[m

flatten(3,2,1)
[0, [1], [[2]], [[[3]]]]
[0,1,2,3]
[0,1,2,[3]]
[0,1,[2],[[3]]]
execution failed with this error:
flatten is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[3:2], .[-5:4], .[:-2], .[-2:], .[3:3][1:], .[10:]]
[0,1,2,3,4,5,6]
[[], [2,3], [0,1,2,3,4], [5,6], [], []]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[.[3:2], .[-5:4], .[:-2], .[-2:], .[3:3][1:], .[10:]]
"abcdefghi"
["","","abcdefg","hi","",""]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

del(.[2:4],.[0],.[-2:])
[0,1,2,3,4,5,6,7]
[1,4,5]
test program: endOfInput context:[]
[31mERROR[m

.[2:4] = ([], ["a","b"], ["a","b","c"])
[0,1,2,3,4,5,6,7]
[0,1,4,5,6,7]
[0,1,"a","b",4,5,6,7]
[0,1,"a","b","c",4,5,6,7]
test program: endOfInput context:[]
[31mERROR[m

reduce range(65540;65536;-1) as $i ([]; .[$i] = $i)|.[65536:]
null
[null,65537,65538,65539,65540]
test program: endOfInput context:[]
[31mERROR[m

1 as $x | 2 as $y | [$x,$y,$x]
null
[1,2,1]
[32mPASS[m

[1,2,3][] as $x | [[4,5,6,7][$x]]
null
[5]
[6]
[7]
[32mPASS[m

42 as $x | . | . | . + 432 | $x + 1
34324
43
[32mPASS[m

1 as $x | [$x,$x,$x as $x | $x]
null
[1,1,1]
[32mPASS[m

[1, {c:3, d:4}] as [$a, {c:$b, b:$c}] | $a, $b, $c
null
1
3
null
test program: endOfInput context:[]
[31mERROR[m

. as {as: $kw, "str": $str, ("e"+"x"+"p"): $exp} | [$kw, $str, $exp]
{"as": 1, "str": 2, "exp": 3}
[1, 2, 3]
test program: endOfInput context:[]
[31mERROR[m

.[] as [$a, $b] | [$b, $a]
[[1], [1, 2, 3]]
[null, 1]
[2, 1]
test program: endOfInput context:[]
[31mERROR[m

. as $i | . as [$i] | $i
[0]
0
test program: endOfInput context:[]
[31mERROR[m

. as [$i] | . as $i | $i
[0]
[0]
test program: endOfInput context:[]
[31mERROR[m

%%FAIL IGNORE MSG
. as [] | null
jq: error: syntax error, unexpected ']', expecting '$' or '[' or '{' (Unix shell quoting issues?) at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

%%FAIL IGNORE MSG
. as {} | null
jq: error: syntax error, unexpected '}' (Unix shell quoting issues?) at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

1+1
null
2
[32mPASS[m

1+1
"wtasdf"
2.0
[32mPASS[m

2-1
null
1
[32mPASS[m

2-(-1)
null
3
[32mPASS[m

1e+0+0.001e3
"I wonder what this will be?"
20e-1
[32mPASS[m

.+4
15
19.0
[32mPASS[m

.+null
{"a":42}
{"a":42}
[32mPASS[m

null+.
null
null
[32mPASS[m

.a+.b
{"a":42}
42
[32mPASS[m

[1,2,3] + [.]
null
[1,2,3,null]
[32mPASS[m

{"a":1} + {"b":2} + {"c":3}
"asdfasdf"
{"a":1, "b":2, "c":3}
[32mPASS[m

"asdf" + "jkl;" + . + . + .
"some string"
"asdfjkl;some stringsome stringsome string"
[32mPASS[m

"\u0000\u0020\u0000" + .
"\u0000\u0020\u0000"
"\u0000 \u0000\u0000 \u0000"
[32mPASS[m

42 - .
11
31
[32mPASS[m

[1,2,3,4,1] - [.,3]
1
[2,4]
[32mPASS[m

[10 * 20, 20 / .]
4
[200, 5]
[32mPASS[m

1 + 2 * 2 + 10 / 2
null
10
[32mPASS[m

[16 / 4 / 2, 16 / 4 * 2, 16 - 4 - 2, 16 - 4 + 2]
null
[2, 8, 10, 14]
[32mPASS[m

25 % 7
null
4
[32mPASS[m

49732 % 472
null
172
[32mPASS[m

1 + tonumber + ("10" | tonumber)
4
15
[32mPASS[m

[{"a":42},.object,10,.num,false,true,null,"b",[1,4]] | .[] as $x | [$x == .[]]
{"object": {"a":42}, "num":10.0}
[true,  true,  false, false, false, false, false, false, false]
[true,  true,  false, false, false, false, false, false, false]
[false, false, true,  true,  false, false, false, false, false]
[false, false, true,  true,  false, false, false, false, false]
[false, false, false, false, true,  false, false, false, false]
[false, false, false, false, false, true,  false, false, false]
[false, false, false, false, false, false, true,  false, false]
[false, false, false, false, false, false, false, true,  false]
[false, false, false, false, false, false, false, false, true ]
[32mPASS[m

[.[] | length]
[[], {}, [1,2], {"a":42}, "asdf", "\u03bc"]
[0, 0, 2, 1, 4, 1]
[32mPASS[m

utf8bytelength
"asdf\u03bc"
6
execution failed with this error:
utf8bytelength is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[] | try utf8bytelength catch .]
[[], {}, [1,2], 55, true, false]
["array ([]) only strings have UTF-8 byte length","object ({}) only strings have UTF-8 byte length","array ([1,2]) only strings have UTF-8 byte length","number (55) only strings have UTF-8 byte length","boolean (true) only strings have UTF-8 byte length","boolean (false) only strings have UTF-8 byte length"]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

map(keys)
[{}, {"abcd":1,"abc":2,"abcde":3}, {"x":1, "z": 3, "y":2}]
[[], ["abc","abcd","abcde"], ["x","y","z"]]
[32mPASS[m

[1,2,empty,3,empty,4]
null
[1,2,3,4]
[32mPASS[m

map(add)
[[], [1,2,3], ["a","b","c"], [[3],[4,5],[6]], [{"a":1}, {"b":2}, {"a":3}]]
[null, 6, "abc", [3,4,5,6], {"a":3, "b": 2}]
[32mPASS[m

map_values(.+1)
[0,1,2]
[1,2,3]
execution failed with this error:
map_values is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

def f: . + 1; def g: def g: . + 100; f | g | f; (f | g), g
3.0
106.0
105.0
test program: endOfInput context:[]
[31mERROR[m

def f: (1000,2000); f
123412345
1000
2000
test program: endOfInput context:[]
[31mERROR[m

def f(a;b;c;d;e;f): [a+1,b,c,d,e,f]; f(.[0];.[1];.[0];.[0];.[0];.[0])
[1,2]
[2,2,1,1,1,1]
test program: endOfInput context:[]
[31mERROR[m

def f: 1; def g: f, def f: 2; def g: 3; f, def f: g; f, g; def f: 4; [f, def f: g; def g: 5; f, g]+[f,g]
null
[4,1,2,3,3,5,4,1,2,3,3]
test program: endOfInput context:[]
[31mERROR[m

def a: 0; . | a
null
0
test program: endOfInput context:[]
[31mERROR[m

def f(a;b;c;d;e;f;g;h;i;j): [j,i,h,g,f,e,d,c,b,a]; f(.[0];.[1];.[2];.[3];.[4];.[5];.[6];.[7];.[8];.[9])
[0,1,2,3,4,5,6,7,8,9]
[9,8,7,6,5,4,3,2,1,0]
test program: endOfInput context:[]
[31mERROR[m

([1,2] + [4,5])
[1,2,3]
[1,2,4,5]
[32mPASS[m

true
[1]
true
[32mPASS[m

null,1,null
"hello"
null
1
null
[32mPASS[m

[1,2,3]
[5,6]
[1,2,3]
[32mPASS[m

[.[]|floor]
[-1.1,1.1,1.9]
[-2, 1, 1]
[32mPASS[m

[.[]|sqrt]
[4,9]
[2,3]
[32mPASS[m

(add / length) as $m | map((. - $m) as $d | $d * $d) | add / length | sqrt
[2,4,4,4,5,5,7,9]
2
[32mPASS[m

atan * 4 * 1000000|floor / 1000000
1
3.141592
execution failed with this error:
atan is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[(3.141592 / 2) * (range(0;20) / 20)|cos * 1000000|floor / 1000000]
null
[1,0.996917,0.987688,0.972369,0.951056,0.923879,0.891006,0.85264,0.809017,0.760406,0.707106,0.649448,0.587785,0.522498,0.45399,0.382683,0.309017,0.233445,0.156434,0.078459]
execution failed with this error:
cos is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[(3.141592 / 2) * (range(0;20) / 20)|sin * 1000000|floor / 1000000]
null
[0,0.078459,0.156434,0.233445,0.309016,0.382683,0.45399,0.522498,0.587785,0.649447,0.707106,0.760405,0.809016,0.85264,0.891006,0.923879,0.951056,0.972369,0.987688,0.996917]
execution failed with this error:
sin is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

def f(x): x | x; f([.], . + [42])
[1,2,3]
[[[1,2,3]]]
[[1,2,3],42]
[[1,2,3,42]]
[1,2,3,42,42]
test program: endOfInput context:[]
[31mERROR[m

def f: .+1; def g: f; def f: .+100; def f(a):a+.+11; [(g|f(20)), f]
1
[33,101]
test program: endOfInput context:[]
[31mERROR[m

def id(x):x; 2000 as $x | def f(x):1 as $x | id([$x, x, x]); def g(x): 100 as $x | f($x,$x+x); g($x)
"more testing"
[1,100,2100.0,100,2100.0]
test program: endOfInput context:[]
[31mERROR[m

def x(a;b): a as $a | b as $b | $a + $b; def y($a;$b): $a + $b; def check(a;b): [x(a;b)] == [y(a;b)]; check(.[];.[]*2)
[1,2,3]
true
test program: endOfInput context:[]
[31mERROR[m

[[20,10][1,0] as $x | def f: (100,200) as $y | def g: [$x + $y, .]; . + $x | g; f[0] | [f][0][1] | f]
999999999
[[110.0, 130.0], [210.0, 130.0], [110.0, 230.0], [210.0, 230.0], [120.0, 160.0], [220.0, 160.0], [120.0, 260.0], [220.0, 260.0]]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

def fac: if . == 1 then 1 else . * (. - 1 | fac) end; [.[] | fac]
[1,2,3,4]
[1,2,6,24]
test program: endOfInput context:[]
[31mERROR[m

reduce .[] as $x (0; . + $x)
[1,2,4]
7
[32mPASS[m

reduce .[] as [$i, {j:$j}] (0; . + $i - $j)
[[2,{"j":1}], [5,{"j":3}], [6,{"j":4}]]
5
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

reduce [[1,2,10], [3,4,10]][] as [$i,$j] (0; . + $i * $j)
null
14
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

reduce . as $n (.; .)
null
null
[32mPASS[m

. as {$a, b: [$c, {$d}]} | [$a, $c, $d]
{"a":1, "b":[2,{"d":3}]}
[1,2,3]
test program: endOfInput context:[]
[31mERROR[m

. as {$a, $b:[$c, $d]}| [$a, $b, $c, $d]
{"a":1, "b":[2,{"d":3}]}
[1,[2,{"d":3}],2,{"d":3}]
test program: endOfInput context:[]
[31mERROR[m

.[] | . as {$a, b: [$c, {$d}]} ?// [$a, {$b}, $e] ?// $f | [$a, $b, $c, $d, $e, $f]
[{"a":1, "b":[2,{"d":3}]}, [4, {"b":5, "c":6}, 7, 8, 9], "foo"]
[1, null, 2, 3, null, null]
[4, 5, null, null, 7, null]
[null, null, null, null, null, "foo"]
test program: endOfInput context:[]
[31mERROR[m

.[] | . as {a:$a} ?// {a:$a} ?// {a:$a} | $a
[[3],[4],[5],6]
test program: endOfInput context:[]
[31mERROR[m

.[] as {a:$a} ?// {a:$a} ?// {a:$a} | $a
[[3],[4],[5],6]
test program: endOfInput context:[]
[31mERROR[m

[[3],[4],[5],6][] | . as {a:$a} ?// {a:$a} ?// {a:$a} | $a
null
test program: endOfInput context:[]
[31mERROR[m

[[3],[4],[5],6] | .[] as {a:$a} ?// {a:$a} ?// {a:$a} | $a
null
test program: endOfInput context:[]
[31mERROR[m

.[] | . as {a:$a} ?// {a:$a} ?// $a | $a
[[3],[4],[5],6]
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

.[] as {a:$a} ?// {a:$a} ?// $a | $a
[[3],[4],[5],6]
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

[[3],[4],[5],6][] | . as {a:$a} ?// {a:$a} ?// $a | $a
null
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

[[3],[4],[5],6] | .[] as {a:$a} ?// {a:$a} ?// $a | $a
null
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

.[] | . as {a:$a} ?// $a ?// {a:$a} | $a
[[3],[4],[5],6]
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

.[] as {a:$a} ?// $a ?// {a:$a} | $a
[[3],[4],[5],6]
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

[[3],[4],[5],6][] | . as {a:$a} ?// $a ?// {a:$a} | $a
null
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

[[3],[4],[5],6] | .[] as {a:$a} ?// $a ?// {a:$a} | $a
null
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

.[] | . as $a ?// {a:$a} ?// {a:$a} | $a
[[3],[4],[5],6]
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

.[] as $a ?// {a:$a} ?// {a:$a} | $a
[[3],[4],[5],6]
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

[[3],[4],[5],6][] | . as $a ?// {a:$a} ?// {a:$a} | $a
null
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

[[3],[4],[5],6] | .[] as $a ?// {a:$a} ?// {a:$a} | $a
null
[3]
[4]
[5]
6
test program: endOfInput context:[]
[31mERROR[m

. as $dot|any($dot[];not)
[1,2,3,4,true,false,1,2,3,4,5]
true
execution failed with this error:
any is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

. as $dot|any($dot[];not)
[1,2,3,4,true]
false
execution failed with this error:
any is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

. as $dot|all($dot[];.)
[1,2,3,4,true,false,1,2,3,4,5]
false
execution failed with this error:
all is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

. as $dot|all($dot[];.)
[1,2,3,4,true]
true
execution failed with this error:
all is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

any(true, error; .)
"badness"
true
execution failed with this error:
any is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

all(false, error; .)
"badness"
false
execution failed with this error:
all is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

any(not)
[]
false
execution failed with this error:
any is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

all(not)
[]
true
execution failed with this error:
all is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

any(not)
[false]
true
execution failed with this error:
any is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

all(not)
[false]
true
execution failed with this error:
all is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[any,all]
[]
[false,true]
execution failed with this error:
any is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[any,all]
[true]
[true,true]
execution failed with this error:
any is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[any,all]
[false]
[false,false]
execution failed with this error:
any is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[any,all]
[true,false]
[true,false]
execution failed with this error:
any is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[any,all]
[null,null,true]
[true,false]
execution failed with this error:
any is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

path(.foo[0,1])
null
["foo", 0]
["foo", 1]
execution failed with this error:
path is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

path(.[] | select(.>3))
[1,5,3]
[1]
execution failed with this error:
path is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

path(.)
42
[]
execution failed with this error:
path is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

try path(.a | map(select(.b == 0))) catch .
{"a":[{"b":0}]}
"Invalid path expression with result [{\"b\":0}]"
test program: endOfInput context:[]
[31mERROR[m

try path(.a | map(select(.b == 0)) | .[0]) catch .
{"a":[{"b":0}]}
"Invalid path expression near attempt to access element 0 of [{\"b\":0}]"
test program: endOfInput context:[]
[31mERROR[m

try path(.a | map(select(.b == 0)) | .c) catch .
{"a":[{"b":0}]}
"Invalid path expression near attempt to access element \"c\" of [{\"b\":0}]"
test program: endOfInput context:[]
[31mERROR[m

try path(.a | map(select(.b == 0)) | .[]) catch .
{"a":[{"b":0}]}
"Invalid path expression near attempt to iterate through [{\"b\":0}]"
test program: endOfInput context:[]
[31mERROR[m

path(.a[path(.b)[0]])
{"a":{"b":0}}
["a","b"]
execution failed with this error:
path is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[paths]
[1,[[],{"a":2}]]
[[0],[1],[1,0],[1,1],[1,1,"a"]]
execution failed with this error:
paths is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[leaf_paths]
[1,[[],{"a":2}]]
[[0],[1,1,"a"]]
execution failed with this error:
leaf_paths is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

["foo",1] as $p | getpath($p), setpath($p; 20), delpaths([$p])
{"bar": 42, "foo": ["a", "b", "c", "d"]}
"b"
{"bar": 42, "foo": ["a", 20, "c", "d"]}
{"bar": 42, "foo": ["a", "c", "d"]}
execution failed with this error:
getpath is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

map(getpath([2])), map(setpath([2]; 42)), map(delpaths([[2]]))
[[0], [0,1], [0,1,2]]
[null, null, 2]
[[0,null,42], [0,1,42], [0,1,42]]
[[0], [0,1], [0,1]]
execution failed with this error:
getpath is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

map(delpaths([[0,"foo"]]))
[[{"foo":2, "x":1}], [{"bar":2}]]
[[{"x":1}], [{"bar":2}]]
execution failed with this error:
delpaths is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

["foo",1] as $p | getpath($p), setpath($p; 20), delpaths([$p])
{"bar":false}
null
{"bar":false, "foo": [null, 20]}
{"bar":false}
execution failed with this error:
getpath is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

delpaths([[-200]])
[1,2,3]
[1,2,3]
execution failed with this error:
delpaths is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

try delpaths(0) catch .
{}
"Paths must be specified as an array"
test program: endOfInput context:[]
[31mERROR[m

del(.), del(empty), del((.foo,.bar,.baz) | .[2,3,0]), del(.foo[0], .bar[0], .foo, .baz.bar[0].x)
{"foo": [0,1,2,3,4], "bar": [0,1]}
null
{"foo": [0,1,2,3,4], "bar": [0,1]}
{"foo": [1,4], "bar": [1]}
{"bar": [1]}
execution failed with this error:
del is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

del(.[1], .[-6], .[2], .[-3:9])
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
[0, 3, 5, 6, 9]
test program: endOfInput context:[]
[31mERROR[m

.message = "goodbye"
{"message": "hello"}
{"message": "goodbye"}
execution failed with this error:
_assign is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

.foo = .bar
{"bar":42}
{"foo":42, "bar":42}
execution failed with this error:
_assign is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

.foo |= .+1
{"foo": 42}
{"foo": 43}
execution failed with this error:
_modify is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

.[] += 2, .[] *= 2, .[] -= 2, .[] /= 2, .[] %=2
[1,3,5]
[3,5,7]
[2,6,10]
[-1,1,3]
[0.5, 1.5, 2.5]
[1,1,1]
execution failed with this error:
_modify is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[] % 7]
[-7,-6,-5,-4,-3,-2,-1,0,1,2,3,4,5,6,7]
[0,-6,-5,-4,-3,-2,-1,0,1,2,3,4,5,6,0]
[32mPASS[m

.foo += .foo
{"foo":2}
{"foo":4}
execution failed with this error:
_modify is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

.[0].a |= {"old":., "new":(.+1)}
[{"a":1,"b":2}]
[{"a":{"old":1, "new":2},"b":2}]
execution failed with this error:
_modify is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

def inc(x): x |= .+1; inc(.[].a)
[{"a":1,"b":2},{"a":2,"b":4},{"a":7,"b":8}]
[{"a":2,"b":2},{"a":3,"b":4},{"a":8,"b":8}]
test program: endOfInput context:[]
[31mERROR[m

.[] | try (getpath(["a",0,"b"]) |= 5) catch .
[null,{"b":0},{"a":0},{"a":null},{"a":[0,1]},{"a":{"b":1}},{"a":[{}]},{"a":[{"c":3}]}]
{"a":[{"b":5}]}
{"b":0,"a":[{"b":5}]}
"Cannot index number with number"
{"a":[{"b":5}]}
"Cannot index number with string \"b\""
"Cannot index object with number"
{"a":[{"b":5}]}
{"a":[{"c":3,"b":5}]}
test program: endOfInput context:[]
[31mERROR[m

.[2][3] = 1
[4]
[4, null, [null, null, null, 1]]
execution failed with this error:
_assign is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

.foo[2].bar = 1
{"foo":[11], "bar":42}
{"foo":[11,null,{"bar":1}], "bar":42}
execution failed with this error:
_assign is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

try ((map(select(.a == 1))[].b) = 10) catch .
[{"a":0},{"a":1}]
"Invalid path expression near attempt to iterate through [{\"a\":1}]"
test program: endOfInput context:[]
[31mERROR[m

try ((map(select(.a == 1))[].a) |= .+1) catch .
[{"a":0},{"a":1}]
"Invalid path expression near attempt to iterate through [{\"a\":1}]"
test program: endOfInput context:[]
[31mERROR[m

def x: .[1,2]; x=10
[0,1,2]
[0,10,10]
test program: endOfInput context:[]
[31mERROR[m

try (def x: reverse; x=10) catch .
[0,1,2]
"Invalid path expression with result [2,1,0]"
test program: endOfInput context:[]
[31mERROR[m

.[] = 1
[1,null,Infinity,-Infinity,NaN,-NaN]
[1,1,1,1,1,1]
test input: Failed reading: not a valid json value context:["json list value"]
[31mERROR[m

[.[] | if .foo then "yep" else "nope" end]
[{"foo":0},{"foo":1},{"foo":[]},{"foo":true},{"foo":false},{"foo":null},{"foo":"foo"},{}]
["yep","yep","yep","yep","nope","nope","yep","nope"]
[32mPASS[m

[.[] | if .baz then "strange" elif .foo then "yep" else "nope" end]
[{"foo":0},{"foo":1},{"foo":[]},{"foo":true},{"foo":false},{"foo":null},{"foo":"foo"},{}]
["yep","yep","yep","yep","nope","nope","yep","nope"]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[if 1,null,2 then 3 else 4 end]
null
[3,4,3]
[32mPASS[m

[if empty then 3 else 4 end]
null
[]
[32mPASS[m

[if 1 then 3,4 else 5 end]
null
[3,4]
[32mPASS[m

[if null then 3 else 5,6 end]
null
[5,6]
[32mPASS[m

[if true then 3 end]
7
[3]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[if false then 3 end]
7
[7]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[if false then 3 else . end]
7
[7]
[32mPASS[m

[.[] | [.foo[] // .bar]]
[{"foo":[1,2], "bar": 42}, {"foo":[1], "bar": null}, {"foo":[null,false,3], "bar": 18}, {"foo":[], "bar":42}, {"foo": [null,false,null], "bar": 41}]
[[1,2], [1], [3], [42], [41]]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

.[] //= .[0]
["hello",true,false,[false],null]
["hello",true,"hello",[false],"hello"]
test program: endOfInput context:[]
[31mERROR[m

.[] | [.[0] and .[1], .[0] or .[1]]
[[true,[]], [false,1], [42,null], [null,false]]
[true,true]
[false,true]
[false,true]
[false,false]
[32mPASS[m

[.[] | not]
[1,0,false,null,true,"hello"]
[false,false,true,true,false,false]
[32mPASS[m

[10 > 0, 10 > 10, 10 > 20, 10 < 0, 10 < 10, 10 < 20]
{}
[true,false,false,false,false,true]
[32mPASS[m

[10 >= 0, 10 >= 10, 10 >= 20, 10 <= 0, 10 <= 10, 10 <= 20]
{}
[true,true,false,false,true,true]
[32mPASS[m

[ 10 == 10, 10 != 10, 10 != 11, 10 == 11]
{}
[true,false,true,false]
[32mPASS[m

["hello" == "hello", "hello" != "hello", "hello" == "world", "hello" != "world" ]
{}
[true,false,false,true]
[32mPASS[m

[[1,2,3] == [1,2,3], [1,2,3] != [1,2,3], [1,2,3] == [4,5,6], [1,2,3] != [4,5,6]]
{}
[true,false,false,true]
[32mPASS[m

[{"foo":42} == {"foo":42},{"foo":42} != {"foo":42}, {"foo":42} != {"bar":42}, {"foo":42} == {"bar":42}]
{}
[true,false,true,false]
[32mPASS[m

[{"foo":[1,2,{"bar":18},"world"]} == {"foo":[1,2,{"bar":18},"world"]},{"foo":[1,2,{"bar":18},"world"]} == {"foo":[1,2,{"bar":19},"world"]}]
{}
[true,false]
[32mPASS[m

[("foo" | contains("foo")), ("foobar" | contains("foo")), ("foo" | contains("foobar"))]
{}
[true, true, false]
[32mPASS[m

[contains(""), contains("\u0000")]
"\u0000"
[true, true]
[32mPASS[m

[contains(""), contains("a"), contains("ab"), contains("c"), contains("d")]
"ab\u0000cd"
[true, true, true, true, true]
[32mPASS[m

[contains("cd"), contains("b\u0000"), contains("ab\u0000")]
"ab\u0000cd"
[true, true, true]
[32mPASS[m

[contains("b\u0000c"), contains("b\u0000cd"), contains("b\u0000cd")]
"ab\u0000cd"
[true, true, true]
[32mPASS[m

[contains("@"), contains("\u0000@"), contains("\u0000what")]
"ab\u0000cd"
[false, false, false]
[32mPASS[m

[.[]|try if . == 0 then error("foo") elif . == 1 then .a elif . == 2 then empty else . end catch .]
[0,1,2,3]
["foo","Cannot index number with string \"a\"",3]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[.[]|(.a, .a)?]
[null,true,{"a":1}]
[null,null,1,1]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[[.[]|[.a,.a]]?]
[null,true,{"a":1}]
[]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

try error("\($__loc__)") catch .
null
"{\"file\":\"<top-level>\",\"line\":1}"
test program: endOfInput context:[]
[31mERROR[m

[.[]|startswith("foo")]
["fo", "foo", "barfoo", "foobar", "barfoob"]
[false, true, false, true, false]
execution failed with this error:
startswith is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[]|endswith("foo")]
["fo", "foo", "barfoo", "foobar", "barfoob"]
[false, true, true, false, false]
execution failed with this error:
endswith is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[] | split(", ")]
["a,b, c, d, e,f",", a,b, c, d, e,f, "]
[["a,b","c","d","e,f"],["","a,b","c","d","e,f",""]]
[32mPASS[m

split("")
"abc"
["a","b","c"]
execution failed with this error:
Data.Text.splitOn: empty input
CallStack (from HasCallStack):
  error, called at libraries/text/Data/Text.hs:1853:18 in text-1.2.3.1:Data.Text
[31mERROR[m

[.[]|ltrimstr("foo")]
["fo", "foo", "barfoo", "foobar", "afoo"]
["fo","","barfoo","bar","afoo"]
execution failed with this error:
ltrimstr is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[]|rtrimstr("foo")]
["fo", "foo", "barfoo", "foobar", "foob"]
["fo","","bar","foobar","foob"]
execution failed with this error:
rtrimstr is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[(index(","), rindex(",")), indices(",")]
"a,bc,def,ghij,klmno"
[1,13,[1,4,8,13]]
[32mPASS[m

indices(1)
[0,1,1,2,3,4,1,5]
[1,2,6]
[32mPASS[m

indices([1,2])
[0,1,2,3,1,4,2,5,1,2,6,7]
[1,8]
was expected, but instead this is the output
[]
[31mFAIL[m

indices([1,2])
[1]
[]
[32mPASS[m

indices(", ")
"a,b, cd,e, fgh, ijkl"
[3,9,14]
[32mPASS[m

[.[]|split(",")]
["a, bc, def, ghij, jklmn, a,b, c,d, e,f", "a,b,c,d, e,f,g,h"]
[["a"," bc"," def"," ghij"," jklmn"," a","b"," c","d"," e","f"],["a","b","c","d"," e","f","g","h"]]
[32mPASS[m

[.[]|split(", ")]
["a, bc, def, ghij, jklmn, a,b, c,d, e,f", "a,b,c,d, e,f,g,h"]
[["a","bc","def","ghij","jklmn","a,b","c,d","e,f"],["a,b,c,d","e,f,g,h"]]
[32mPASS[m

[.[] * 3]
["a", "ab", "abc"]
["aaa", "ababab", "abcabcabc"]
[32mPASS[m

[.[] / ","]
["a, bc, def, ghij, jklmn, a,b, c,d, e,f", "a,b,c,d, e,f,g,h"]
[["a"," bc"," def"," ghij"," jklmn"," a","b"," c","d"," e","f"],["a","b","c","d"," e","f","g","h"]]
[32mPASS[m

[.[] / ", "]
["a, bc, def, ghij, jklmn, a,b, c,d, e,f", "a,b,c,d, e,f,g,h"]
[["a","bc","def","ghij","jklmn","a,b","c,d","e,f"],["a,b,c,d","e,f,g,h"]]
[32mPASS[m

map(.[1] as $needle | .[0] | contains($needle))
[[[],[]], [[1,2,3], [1,2]], [[1,2,3], [3,1]], [[1,2,3], [4]], [[1,2,3], [1,4]]]
[true, true, true, false, false]
execution failed with this error:
Not yet implemented: containement on array
CallStack (from HasCallStack):
  error, called at jq.hs:84:7 in main:Main
[31mERROR[m

map(.[1] as $needle | .[0] | contains($needle))
[[["foobar", "foobaz"], ["baz", "bar"]], [["foobar", "foobaz"], ["foo"]], [["foobar", "foobaz"], ["blap"]]]
[true, true, false]
execution failed with this error:
Not yet implemented: containement on array
CallStack (from HasCallStack):
  error, called at jq.hs:84:7 in main:Main
[31mERROR[m

[({foo: 12, bar:13} | contains({foo: 12})), ({foo: 12} | contains({})), ({foo: 12, bar:13} | contains({baz:14}))]
{}
[true, true, false]
execution failed with this error:
Not yet implemented: containement on object
CallStack (from HasCallStack):
  error, called at jq.hs:84:7 in main:Main
[31mERROR[m

{foo: {baz: 12, blap: {bar: 13}}, bar: 14} | contains({bar: 14, foo: {blap: {}}})
{}
true
execution failed with this error:
Not yet implemented: containement on object
CallStack (from HasCallStack):
  error, called at jq.hs:84:7 in main:Main
[31mERROR[m

{foo: {baz: 12, blap: {bar: 13}}, bar: 14} | contains({bar: 14, foo: {blap: {bar: 14}}})
{}
false
execution failed with this error:
Not yet implemented: containement on object
CallStack (from HasCallStack):
  error, called at jq.hs:84:7 in main:Main
[31mERROR[m

sort
[42,[2,5,3,11],10,{"a":42,"b":2},{"a":42},true,2,[2,6],"hello",null,[2,5,6],{"a":[],"b":1},"abc","ab",[3,10],{},false,"abcd",null]
[null,null,false,true,2,10,42,"ab","abc","abcd","hello",[2,5,3,11],[2,5,6],[2,6],[3,10],{},{"a":42},{"a":42,"b":2},{"a":[],"b":1}]
[32mPASS[m

(sort_by(.b) | sort_by(.a)), sort_by(.a, .b), sort_by(.b, .c), group_by(.b), group_by(.a + .b - .c == 2)
[{"a": 1, "b": 4, "c": 14}, {"a": 4, "b": 1, "c": 3}, {"a": 1, "b": 4, "c": 3}, {"a": 0, "b": 2, "c": 43}]
[{"a": 0, "b": 2, "c": 43}, {"a": 1, "b": 4, "c": 14}, {"a": 1, "b": 4, "c": 3}, {"a": 4, "b": 1, "c": 3}]
[{"a": 0, "b": 2, "c": 43}, {"a": 1, "b": 4, "c": 14}, {"a": 1, "b": 4, "c": 3}, {"a": 4, "b": 1, "c": 3}]
[{"a": 4, "b": 1, "c": 3}, {"a": 0, "b": 2, "c": 43}, {"a": 1, "b": 4, "c": 3}, {"a": 1, "b": 4, "c": 14}]
[[{"a": 4, "b": 1, "c": 3}], [{"a": 0, "b": 2, "c": 43}], [{"a": 1, "b": 4, "c": 14}, {"a": 1, "b": 4, "c": 3}]]
[[{"a": 1, "b": 4, "c": 14}, {"a": 0, "b": 2, "c": 43}], [{"a": 4, "b": 1, "c": 3}, {"a": 1, "b": 4, "c": 3}]]
execution failed with this error:
sort_by is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

unique
[1,2,5,3,5,3,1,3]
[1,2,3,5]
[32mPASS[m

unique
[]
[]
[32mPASS[m

[min, max, min_by(.[1]), max_by(.[1]), min_by(.[2]), max_by(.[2])]
[[4,2,"a"],[3,1,"a"],[2,4,"a"],[1,3,"a"]]
[[1,3,"a"],[4,2,"a"],[3,1,"a"],[2,4,"a"],[4,2,"a"],[1,3,"a"]]
execution failed with this error:
min_by is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[min,max,min_by(.),max_by(.)]
[]
[null,null,null,null]
execution failed with this error:
min_by is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

.foo[.baz]
{"foo":{"bar":4},"baz":"bar"}
4
[32mPASS[m

.[] | .error = "no, it's OK"
[{"error":true}]
{"error": "no, it's OK"}
execution failed with this error:
_assign is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[{a:1}] | .[] | .a=999
null
{"a": 999}
execution failed with this error:
_assign is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

to_entries
{"a": 1, "b": 2}
[{"key":"a", "value":1}, {"key":"b", "value":2}]
[32mPASS[m

from_entries
[{"key":"a", "value":1}, {"Key":"b", "Value":2}, {"name":"c", "value":3}, {"Name":"d", "Value":4}]
{"a": 1, "b": 2, "c": 3, "d": 4}
execution failed with this error:
Cannot use null as object key
CallStack (from HasCallStack):
  error, called at jq.hs:84:7 in main:Main
[31mERROR[m

with_entries(.key |= "KEY_" + .)
{"a": 1, "b": 2}
{"KEY_a": 1, "KEY_b": 2}
execution failed with this error:
_modify is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

map(has("foo"))
[{"foo": 42}, {}]
[true, false]
[32mPASS[m

map(has(2))
[[0,1], ["a","b","c"]]
[false, true]
[32mPASS[m

keys
[42,3,35]
[0,1,2]
[32mPASS[m

[][.]
1000000000000000000
null
[32mPASS[m

map([1,2][0:.])
[-1, 1, 2, 3, 1000000000000000000]
[[1], [1], [1,2], [1,2], [1,2]]
test program: endOfInput context:[]
[31mERROR[m

{"k": {"a": 1, "b": 2}} * .
{"k": {"a": 0,"c": 3}}
{"k": {"a": 0, "b": 2, "c": 3}}
execution failed with this error:
object and object cannot be multiplied
CallStack (from HasCallStack):
  error, called at jq.hs:84:7 in main:Main
[31mERROR[m

{"k": {"a": 1, "b": 2}, "hello": {"x": 1}} * .
{"k": {"a": 0,"c": 3}, "hello": 1}
{"k": {"a": 0, "b": 2, "c": 3}, "hello": 1}
execution failed with this error:
object and object cannot be multiplied
CallStack (from HasCallStack):
  error, called at jq.hs:84:7 in main:Main
[31mERROR[m

{"k": {"a": 1, "b": 2}, "hello": 1} * .
{"k": {"a": 0,"c": 3}, "hello": {"x": 1}}
{"k": {"a": 0, "b": 2, "c": 3}, "hello": {"x": 1}}
execution failed with this error:
object and object cannot be multiplied
CallStack (from HasCallStack):
  error, called at jq.hs:84:7 in main:Main
[31mERROR[m

{"a": {"b": 1}, "c": {"d": 2}, "e": 5} * .
{"a": {"b": 2}, "c": {"d": 3, "f": 9}}
{"a": {"b": 2}, "c": {"d": 3, "f": 9}, "e": 5}
execution failed with this error:
object and object cannot be multiplied
CallStack (from HasCallStack):
  error, called at jq.hs:84:7 in main:Main
[31mERROR[m

[.[]|arrays]
[1,2,"foo",[],[3,[]],{},true,false,null]
[[],[3,[]]]
execution failed with this error:
arrays is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[]|objects]
[1,2,"foo",[],[3,[]],{},true,false,null]
[{}]
execution failed with this error:
objects is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[]|iterables]
[1,2,"foo",[],[3,[]],{},true,false,null]
[[],[3,[]],{}]
execution failed with this error:
iterables is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[]|scalars]
[1,2,"foo",[],[3,[]],{},true,false,null]
[1,2,"foo",true,false,null]
execution failed with this error:
scalars is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[]|values]
[1,2,"foo",[],[3,[]],{},true,false,null]
[1,2,"foo",[],[3,[]],{},true,false]
execution failed with this error:
values is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[]|booleans]
[1,2,"foo",[],[3,[]],{},true,false,null]
[true,false]
execution failed with this error:
booleans is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[.[]|nulls]
[1,2,"foo",[],[3,[]],{},true,false,null]
[null]
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

flatten
[0, [1], [[2]], [[[3]]]]
[0, 1, 2, 3]
execution failed with this error:
flatten is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

flatten(0)
[0, [1], [[2]], [[[3]]]]
[0, [1], [[2]], [[[3]]]]
execution failed with this error:
flatten is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

flatten(2)
[0, [1], [[2]], [[[3]]]]
[0, 1, 2, [3]]
execution failed with this error:
flatten is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

flatten(2)
[0, [1, [2]], [1, [[3], 2]]]
[0, 1, 2, 1, [3], 2]
execution failed with this error:
flatten is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

try flatten(-1) catch .
[0, [1], [[2]], [[[3]]]]
"flatten depth must not be negative"
test program: endOfInput context:[]
[31mERROR[m

transpose
[[1], [2,3]]
[[1,2],[null,3]]
execution failed with this error:
transpose is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

ascii_upcase
"useful but not for √©"
"USEFUL BUT NOT FOR √©"
execution failed with this error:
ascii_upcase is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

bsearch(0,2,4)
[1,2,3]
-1
1
-4
execution failed with this error:
bsearch is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

strftime("%Y-%m-%dT%H:%M:%SZ")
[2015,2,5,23,51,47,4,63]
"2015-03-05T23:51:47Z"
execution failed with this error:
strftime is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

strftime("%A, %B %d, %Y")
1435677542.822351
"Tuesday, June 30, 2015"
execution failed with this error:
strftime is not defined (arity 2)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

gmtime
1425599507
[2015,2,5,23,51,47,4,63]
execution failed with this error:
gmtime is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

import "a" as foo; import "b" as bar; def fooa: foo::a; [fooa, bar::a, bar::b, foo::a]
null
["a","b","c","a"]
test program: endOfInput context:[]
[31mERROR[m

import "c" as foo; [foo::a, foo::c]
null
[0,"acmehbah"]
test program: endOfInput context:[]
[31mERROR[m

include "c"; [a, c]
null
[0,"acmehbah"]
test program: endOfInput context:[]
[31mERROR[m

import "data" as $e; import "data" as $d; [$d[].this,$e[].that,$d::d[].this,$e::e[].that]|join(";")
null
"is a test;is too;is a test;is too"
test program: endOfInput context:[]
[31mERROR[m

include "shadow1"; e
null
2
test program: endOfInput context:[]
[31mERROR[m

include "shadow1"; include "shadow2"; e
null
3
test program: endOfInput context:[]
[31mERROR[m

import "shadow1" as f; import "shadow2" as f; import "shadow1" as e; [e::e, f::e]
null
[2,3]
test program: endOfInput context:[]
[31mERROR[m

%%FAIL
module (.+1); 0
jq: error: Module metadata must be constant at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

%%FAIL
include "a" (.+1); 0
jq: error: Module metadata must be constant at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

%%FAIL
include "a" []; 0
jq: error: Module metadata must be an object at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

%%FAIL
include "\ "; 0
jq: error: Invalid escape at line 1, column 4 (while parsing '"\ "') at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

%%FAIL
include "\(a)"; 0
jq: error: Import path must be constant at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

modulemeta
"c"
{"whatever":null,"deps":[{"as":"foo","is_data":false,"relpath":"a"},{"search":"./","as":"d","is_data":false,"relpath":"d"},{"search":"./","as":"d2","is_data":false,"relpath":"d"},{"search":"./../lib/jq","as":"e","is_data":false,"relpath":"e"},{"search":"./../lib/jq","as":"f","is_data":false,"relpath":"f"},{"as":"d","is_data":true,"relpath":"data"}]}
execution failed with this error:
modulemeta is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

modulemeta | .deps |= length
"c"
{"whatever":null,"deps":6}
execution failed with this error:
modulemeta is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

%%FAIL IGNORE MSG
import "syntaxerror" as e; .
jq: error: syntax error, unexpected ';', expecting $end (Unix shell quoting issues?) at /home/nico/ws/jq/tests/modules/syntaxerror/syntaxerror.jq, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

%%FAIL IGNORE MSG
%::wat
jq: error: syntax error, unexpected '%', expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

import "test_bind_order" as check; check::check
null
true
test program: endOfInput context:[]
[31mERROR[m

try -. catch .
"very-long-string"
"string (\"very-long-...) cannot be negated"
test program: endOfInput context:[]
[31mERROR[m

join(",")
["1",2,true,false,3.4]
"1,2,true,false,3.4"
[32mPASS[m

.[] | join(",")
[[], [null], [null,null], [null,null,null]]
""
""
","
",,"
[32mPASS[m

.[] | join(",")
[["a",null], [null,"a"]]
"a,"
",a"
[32mPASS[m

try join(",") catch .
["1","2",{"a":{"b":{"c":33}}}]
"string (\"1,2,\") and object ({\"a\":{\"b\":{...) cannot be added"
test program: endOfInput context:[]
[31mERROR[m

try join(",") catch .
["1","2",[3,4,5]]
"string (\"1,2,\") and array ([3,4,5]) cannot be added"
test program: endOfInput context:[]
[31mERROR[m

{if:0,and:1,or:2,then:3,else:4,elif:5,end:6,as:7,def:8,reduce:9,foreach:10,try:11,catch:12,label:13,import:14,include:15,module:16}
null
{"if":0,"and":1,"or":2,"then":3,"else":4,"elif":5,"end":6,"as":7,"def":8,"reduce":9,"foreach":10,"try":11,"catch":12,"label":13,"import":14,"include":15,"module":16}
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

try (1/.) catch .
0
"number (1) and number (0) cannot be divided because the divisor is zero"
test program: endOfInput context:[]
[31mERROR[m

try (1%.) catch .
0
"number (1) and number (0) cannot be divided (remainder) because the divisor is zero"
test program: endOfInput context:[]
[31mERROR[m

%%FAIL
1/0
jq: error: Division by zero? at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

[range(-52;52;1)] as $powers | [$powers[]|pow(2;.)|log2] == $powers
null
true
execution failed with this error:
pow is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

[range(-99/2;99/2;1)] as $orig | [$orig[]|pow(2;.)|log2] as $back | ($orig|keys)[]|. as $k | (($orig|.[$k])-($back|.[$k]))|if . < 0 then . * -1 else . end|select(.>.00005)
null
test program: endOfInput context:[]
[31mERROR[m

%%FAIL IGNORE MSG
}
jq: error: syntax error, unexpected INVALID_CHARACTER, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter","("]
[31mERROR[m

(.[{}] = 0)?
null
test program: endOfInput context:[]
[31mERROR[m

INDEX(range(5)|[., "foo\(.)"]; .[0])
null
{"0":[0,"foo0"],"1":[1,"foo1"],"2":[2,"foo2"],"3":[3,"foo3"],"4":[4,"foo4"]}
execution failed with this error:
INDEX is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

JOIN({"0":[0,"abc"],"1":[1,"bcd"],"2":[2,"def"],"3":[3,"efg"],"4":[4,"fgh"]}; .[0]|tostring)
[[5,"foo"],[3,"bar"],[1,"foobar"]]
[[[5,"foo"],null],[[3,"bar"],[3,"efg"]],[[1,"foobar"],[1,"bcd"]]]
execution failed with this error:
JOIN is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

range(5;10)|IN(range(10))
null
true
true
true
true
true
[32mPASS[m

range(5;13)|IN(range(0;10;3))
null
false
true
false
false
true
false
false
false
[32mPASS[m

range(10;12)|IN(range(10))
null
false
false
[32mPASS[m

IN(range(10;20); range(10))
null
false
[32mPASS[m

IN(range(5;20); range(10))
null
true
[32mPASS[m

(.a as $x | .b) = "b"
{"a":null,"b":null}
{"a":null,"b":"b"}
execution failed with this error:
_assign is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

(.. | select(type == "object" and has("b") and (.b | type) == "array")|.b) |= .[0]
{"a": {"b": [1, {"b": 3}]}}
{"a": {"b": 1}}
test program: Failed reading: satisfyWith context:["filter","as filter","conc filter","simple filter",")"]
[31mERROR[m

isempty(empty)
null
true
[32mPASS[m

isempty(range(3))
null
false
[32mPASS[m

isempty(1,error("foo"))
null
false
[32mPASS[m

index("")
""
null
[32mPASS[m

builtins|length > 10
null
true
execution failed with this error:
builtins is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

"-1"|IN(builtins[] / "/"|.[1])
null
false
execution failed with this error:
builtins is not defined (arity 1)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

all(builtins[] / "/"; .[1]|tonumber >= 0)
null
true
execution failed with this error:
all is not defined (arity 3)
CallStack (from HasCallStack):
  error, called at jq.hs:586:16 in main:Main
[31mERROR[m

builtins|any(.[:1] == "_")
null
false
test program: endOfInput context:[]
[31mERROR[m

